pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/internal/asm/ParseLock.java

```diff
@@ -32,14 +32,22 @@ private ParseStatus getFinalStatus() {
                     try {
                         boolean success = doParse();
                         status = success ? ParseStatus.FULL : ParseStatus.FAILED;
-                        this.status = status;
                         finishParse(!success);
                     } catch (Throwable t) {
                         status = ParseStatus.FAILED;
-                        this.status = status;
-                        LOG.error(t.toString(), t);
+                        LOG.error("Parsing failed in ParseLock#doParse()", t);
```

Comment:
Figured out the 2nd stacktrace of #4757 now, the case when GenericSigBase#supertIfs is null:
- We are parsing a signature of some kind
- This runs into a NPE, because ClassStub#enclosingInfo is null (because we updated ParseLock#status too early)
- This NPE is caught here - and that means, that doParse() didn't finish correctly
- In doParse() we would set the super interfaces
  - https://github.com/pmd/pmd/blob/ce347bd7361bcf75be67133ea967fc2c0d923ff6/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/internal/asm/GenericSigBase.java#L155
  - https://github.com/pmd/pmd/blob/ce347bd7361bcf75be67133ea967fc2c0d923ff6/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/internal/asm/SignatureParser.java#L58
- But two lines earlier in SignatureParser - https://github.com/pmd/pmd/blob/ce347bd7361bcf75be67133ea967fc2c0d923ff6/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/internal/asm/SignatureParser.java#L56
  - we call typeParamsWrapper() that eventually calls
  - ClassStub#getEnclosingTypeParameterOwner() https://github.com/pmd/pmd/blob/ce347bd7361bcf75be67133ea967fc2c0d923ff6/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/internal/asm/ClassStub.java#L373
  - And that's where the NPE happens

So, the issue that superIfs is null, is a consequential error of the earlier NPE and both should be fixed now.

At least, I couldn't reproduce them anymore with Thread.sleep.